components/server/src/billing/entitlement-service.ts

```diff
@@ -81,7 +78,7 @@ export class CommunityEntitlementService implements EntitlementService {
         date: Date,
         runningInstances: Promise<WorkspaceInstance[]>,
     ): Promise<MayStartWorkspaceResult> {
-        return { mayStart: true };
```

Comment:
I'd prefer to keep a readable signal. 
the empty object is hard to understand throughout the chain of calls.

Reply:
But it makes it very likely to mis-interpret. Looking at the interface definition I think makes it very clear that this is a bag of reasons to _not_ start this workspace.

Or do you mean this specific site? This would be `{ outOfCredits: false }`, for instances, which is not really relevant, because this is a purely Chargebee specific field. I'd rather keep it as simple as possible.

Reply:
> Looking at the interface definition I think makes it very clear that this is a bag of reasons to not start this workspace.

Let walk though this by example to show what I mean by a missing readability about this signal
Given
```
let result = entitlementService.mayStartWorkspace()
```
we want to decide if the workspace is allowed to take off. Having no strong signal on this, we need to check the 3  optional attributes to be present. 
This is possible to do so – *now* after applying a [polarizing filter](https://github.com/gitpod-io/gitpod/pull/11936/commits/b0b4d88b9ac884ce8052439849b50c2af6f7d21d#diff-42937a09d5730082001f0989261d91bf1560753c97c60e29ce58c76f0ca7c603R269), which showed already how fragile that can be. 

But think of a future task to extend the `entitlementService.mayStartWorkspace` check with more negative signals. It feels to be error prone to extend the evaluation of a result object to meet the extended computation.